src/liboslexec/batched_analysis.cpp

```diff
@@ -106,6 +106,10 @@ namespace  // Unnamed
 bool
 are_op_results_always_implicitly_varying(ustring opname)
 {
+    OIIO_PRAGMA_WARNING_PUSH
+#if OIIO_CLANG_VERSION >= 140000
```

Comment:
This feels pretty noisy in the source, why not just disable it via the command line flag -Wno-bitwise-instead-of-logical?
Every use of | or & should have been an explicit decision.

Reply:
Yeah, I could do that I suppose.

I wasn't sure of the merits of disabling the warning for the whole codebase, which of course would have the effect of NOT catching any times we accidentally used bitwise ops when we meant logical. That warning, for the vast majority of code, is not a bad idea. I think it serves a good purpose.

On the other hand, we never had the warning before, and now that we've turned it on, the only places it caught were instances where we purposely used the bitwise ops. So maybe it's trying to catch a mistake that we never make in practice, so it's always a false negative.

Let me mull this over (@fpsunflower or others, do you have an opinion?) but I think you have me leaning toward thinking that you are right and I should just disable the warning wholesale.

Reply:
As the warning it self doesn't can't actually determine if you intended to skip the short-circuit behavior of logical || and &&, IMO it shouldn't be on by default.  If tracking down a bug, maybe it would be nice to get such a report of all places bitwise was used to verify they were all intentional, but certainly not as part of a -Werror build configuration.
As a typical use is 
`if (ptr && ptr->foobar())`
If that were just a bitwise &, then both sides will execute and the order is undefined:
`if (ptr & ptr->foobar())`
then a crash might occur.  However, C++ has made it 37 years without this warning,
mostly because the call stack should lead them right to the offending line.  However if not a simple invalid pointer some incorrect behavior of the "order undefined" and "both sides" execute behavior could happen.

The warning itself actually says
`note: cast one or both operands to int to silence this warning`
Cast to an int? But it was already a bool and adding additional casts just to make this warning happy doesn't make sense.  Bitwise | and & on bools is legitimate, and not a cause for warning.  Maybe there should be a -Wlook-for-common-typos flag that does this (and others like assignment inside if statement).

For performance code we often want to skip the short circuit and if an additional branch is generated is a performance degradation, so there might be a trade off between warning for a "potential" typo and performance.

Take a look at example
https://godbolt.org/z/bnKMKnn5K
We can see a `jmp` in the assembly, change MY_OR at the top to bitwise | and it goes away.  Branch prediction penalties and additional management of SIMD masks and mask registers are real costs.  So developer should choose || or | depending on their needs.

Reply:
I'm usually in favor of adopting all warnings, but this one doesn't seem to be fully thought out. I'm fine disabling this one for now.